core/field_dropdown.js

```diff
@@ -611,4 +633,15 @@ Blockly.FieldDropdown.prototype.onBlocklyAction = function(action) {
   return Blockly.FieldDropdown.superClass_.onBlocklyAction.call(this, action);
 };
 
+/**
+ * Dispose of all DOM objects and events belonging to this editable field.
+ * @package
+ * @override
+ */
+Blockly.FieldDropdown.prototype.dispose = function() {
+  Blockly.FieldDropdown.superClass_.dispose.call(this);
+  this.imageElement_ = null;
```

Comment:
As far as I know, you don't need to set these to null in this case.

Since fields don't get disposed of directly, and are only disposed through input.removeField the input handles dropping the reference to the field. Because the field is then unreachable, these properties become unreachable as well, so the elements get disposed of, instead of sticking around as detached nodes.

Might be useful to @NeilFraser for an opinion (and also to check my reasoning).

Reply:
I think you might be right, did a quick test and it doesn't appear to leave any detached nodes behind when not set to null.